### PR TITLE
Optimize `GET`ting and object's payload

### PR DESCRIPTION
**Pull request by paolobarbolini**
Removes a memcpy and an allocation from the object read implementation

### CODE REVIEW

File: async-nats/src/jetstream/object_store/mod.rs [hunk 1]
```diff
@@ -568,7 +569,7 @@ impl Stream for List<'_> {
 /// Represents an object stored in a bucket.
 pub struct Object<'a> {
     pub info: ObjectInfo,
-    remaining_bytes: Vec<u8>,
+    remaining_bytes: VecDeque<u8>,
```
Comment (caspervonb) [inline async-nats/src/jetstream/object_store/mod.rs:572]: What do you think about using a bytes buffer here instead?

**Review by caspervonb — COMMENTED** [review]
Was wondering if we could get away with having remaiing_bytes as a `BytesMut` buffer here.

**Comment by paolobarbolini** [comment]
> Was wondering if we could get away with having remaiing_bytes as a `BytesMut` buffer here.

It could work but if you don't need a `Bytes` at the end you're just wasting time with the ref-counting and all of the internal `bytes` machinery

**Review by caspervonb — APPROVED** [review]
lgtm

**Review by Jarema — APPROVED** [review]
LGTM!

**Pull request merged**